src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4121,18 +4121,31 @@ public bool SendToAsync(SocketAsyncEventArgs e)
 
             // Prepare for and make the native call.
             e.StartOperationCommon(this, SocketAsyncOperation.SendTo);
+
+            EndPoint oldEndPoint = _rightEndPoint;
+            if (_rightEndPoint == null)
+            {
+                _rightEndPoint = endPointSnapshot;
+            }
+
             SocketError socketError;
             try
             {
                 socketError = e.DoOperationSendTo(_handle);
             }
             catch
             {
+                _rightEndPoint = null;
                 // Clear in-use flag on event args object.
                 e.Complete();
                 throw;
             }
 
+            if (!CheckErrorAndUpdateStatus(socketError))
+            {
```

Comment:
I have no idea what black magic could trigger the catch block here, on the other hand `socketError == AccessDenied` is something we can actually hit, see my test cases. Like in `DoBeginSendTo` I'm resetting `_rightEndPoint` when hitting a non-success & non-pending case.